### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by meho99**

### CODE REVIEW

File: contracts/helpers/BaseAdminUpgradeabilityProxy.sol [hunk 1]
```diff
@@ -16,114 +16,114 @@ import "@openzeppelin/upgrades/contracts/upgradeability/UpgradeabilityProxy.sol"
  * feature proposal that would enable this to be done automatically.
  */
 contract BaseAdminUpgradeabilityProxy is BaseUpgradeabilityProxy {
-  /**
```
Comment (dharkmattr) [inline contracts/helpers/BaseAdminUpgradeabilityProxy.sol:19]: What changed in this file?

File: contracts/masset/BasketManagerV3.sol [hunk 2]
```diff
@@ -174,6 +187,7 @@ contract BasketManagerV3 is InitializableOwnable {
 
     function setFactor(address _basset, int256 _factor) public validBasset(_basset) onlyOwner {
         require(_factor != 0, "invalid factor");
+        require(_factor == 1 || _factor % 10 == 0, "factor must be power of 10");
```
Comment (dharkmattr) [inline contracts/masset/BasketManagerV3.sol:190]: This doesn't check that factor is power of 10, only that it is a factor of 10

File: contracts/masset/BasketManagerV3.sol [hunk 3]
```diff
@@ -182,7 +196,7 @@ contract BasketManagerV3 is InitializableOwnable {
     }
 
     function setPaused(address _basset, bool _flag) public validBasset(_basset) onlyOwner {
-        pausedMap[_basset] == _flag;
+        pausedMap[_basset] = _flag;
```
Comment (dharkmattr) [inline contracts/masset/BasketManagerV3.sol:199]: good catch!

File: contracts/masset/BasketManagerV3.sol [hunk 4]
```diff
@@ -52,7 +53,11 @@ contract BasketManagerV3 is InitializableOwnable {
         uint256 _bassetQuantity) public view validBasset(_basset) notPaused(_basset) returns(bool) {
 
         uint256 massetQuantity = convertBassetToMassetQuantity(_basset, _bassetQuantity);
-        uint256 balance = IERC20(_basset).balanceOf(masset).add(massetQuantity);
+        uint256 bassetBalance = IERC20(_basset).balanceOf(masset);
```
Comment (dharkmattr) [inline contracts/masset/BasketManagerV3.sol:56]: yes! otherwise precision is not calculated correctly.

**Review by dharkmattr — APPROVED** [review]

File: contracts/helpers/BaseAdminUpgradeabilityProxy.sol (hunk 1 above)
Comment (meho99) [reply contracts/helpers/BaseAdminUpgradeabilityProxy.sol:19]: I changed it because of linter warnings that there should be 4 spaces instead of 2.

Comment (meho99) [reply contracts/helpers/BaseAdminUpgradeabilityProxy.sol:19]: Github is showing the diff in some weird way for some reason

**Pull request merged**